### ISSUE TITLE
JSON response screen post completion of Survey

### ISSUE THREAD

**Issue by alokomkar** [SurveyMonkey/surveymonkey-android-sdk#61]
Hi,

We are trying to integrate survey via SurveyMonkey in our app. We have a couple of surveys created in the dashboard with different hash ids.

We have trigger points for survey at specific user action in the app. The survey is showing up repeatedly without regard to interval time. Here's our code in BaseActivity : 

```
 private val SURVEY_MONKEY_REQUEST_CODE = 4324
 private var mSurveyHashId: String = ""

    fun showSurvey(surveyHashId: String): Boolean {
        val isSameSurvey = mSurveyHashId != surveyHashId
        if (isSameSurvey) {
            mSurveyHashId = surveyHashId
            mSurveyMonkeyHelper = mSurveyMap[mSurveyHashId]
            mSurveyMonkeyHelper?.apply {
                showSurvey()
            }
        }
        return isSameSurvey
    }

    fun forceShowSurvey( surveyHashId: String ) {
        SurveyMonkeyHelper(this, SURVEY_MONKEY_REQUEST_CODE, surveyHashId).showSurvey()
    }

    private var mSurveyMonkeyHelper: SurveyMonkeyHelper? = null
    private val mSurveyMap = HashMap<String, SurveyMonkeyHelper>()

    fun initializeSurvey(surveyHashId: String) {
        mSurveyMap[surveyHashId] = SurveyMonkeyHelper(this, SURVEY_MONKEY_REQUEST_CODE, surveyHashId)
        mSurveyMonkeyHelper = mSurveyMap[surveyHashId]
    }

   override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == SURVEY_MONKEY_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
            //Do something with the result.
            if (mSurveyHashId == SurveyMonkeyHelper.HASH_COLLECTIVE_CHECKOPUT) {
                supportFinishAfterTransition()
            }
        }
    }
```
SurveyMonkeyHelper class :

```
class SurveyMonkeyHelper(private val activity: Activity, val requestCode: Int, private val hashKey: String) {

    private val sdkInstance = SurveyMonkey()

    init {

        //No of days between the surveys
        val days = when (hashKey) {
            HASH_KEY_NPS -> 90
            HASH_ACHIEVEMENT_CODE_CLAIMED -> 300
            HASH_NON_SAVERS -> 30
            HASH_SPEND_TO_SAVE -> 30
            HASH_COLLECTIVE_CHECKOPUT -> 30
            else -> 90
        }
        sdkInstance.onStart(
                activity,
                requestCode,
                hashKey,
                activity.getString(R.string.app_name),
                "",
                0L,
                TimeUnit.MILLISECONDS.convert(days.toLong(), TimeUnit.DAYS),
                TimeUnit.MILLISECONDS.convert(days.toLong(), TimeUnit.DAYS),
                null)
    }

    fun showSurvey() {
        sdkInstance.startSMFeedbackActivityForResult(activity, requestCode, hashKey)
    }

    companion object {
       //Survey ids changed.
        const val HASH_KEY_NPS = "312412"
        const val HASH_ACHIEVEMENT_CODE_CLAIMED = "123414"
        const val HASH_NON_SAVERS = "234124"
        const val HASH_SPEND_TO_SAVE = "123412"
        const val HASH_COLLECTIVE_CHECKOPUT = "1324124"
    }
}
```
Let us know if we are missing something? 

Second issue : There's a screen which appears on the phone post a survey is completed with some json formatted text. How to handle this?

Update : 
1. We were able to solve interval time issue by only calling sdkInstance.onStart()
2. Issue : We are getting a Masher API key in the json response screen, this is an unintended behaviour.

@benleiken 

Here's the response we see : 

{ "mashery_api_key" : "some_key_string",
  "respondent_token" : "some_token_string"...

}

**Comment by klaasopurk** [comment]
Same problem here. After submit the form, for a few seconds the response is visible. We don't know if this could give a bigger problem. 
Are you sure that the response can not be misused to request or change other data?

**Comment by ayeung** [comment]
Will be fixed in next release: 1.0.7

**Comment by ayeung** [comment]
Just released 2.0.0.  Check it out when you get the chance.  I'll be closing this issue for now, feel free to reopen and/or comment if you are still having issues.

Note that I did run into a situation where the previous version of the library was cached on Android Studio, so I would also invalidate the cache to make sure you are grabbing the latest if you run into any issues.
